backend/src/apiserver/server/api_converter_test.go

```diff
@@ -112,6 +133,63 @@ func TestToApiRunDetail(t *testing.T) {
 			Status:       "running",
 			PipelineSpec: &api.PipelineSpec{
 				WorkflowManifest: "manifest",
+				RuntimeConfig: &api.PipelineSpec_RuntimeConfig{
+					Parameters:   v2RuntimeParams,
+					PipelineRoot: "model-pipeline-root",
+				},
+			},
+			ResourceReferences: []*api.ResourceReference{
+				{Key: &api.ResourceKey{Type: api.ResourceType_JOB, Id: "job123"},
+					Name: "j123", Relationship: api.Relationship_CREATOR},
+			},
+		},
+		PipelineRuntime: &api.PipelineRuntime{
+			WorkflowManifest: "workflow123",
+		},
+	}
+	assert.Equal(t, expectedApiRun.String(), apiRun.String())
```

Comment:
no need for `Sting()`

Reply:
Added explanation for usage of `String()`. Some fields in ApiRuns are only used for protobuff, and may be different although the fields we use are the same. Using string representation avoids the issue.